adaptive/learner/sequence_learner.py

```diff
@@ -0,0 +1,132 @@
+import sys
+from copy import copy
+
+from sortedcontainers import SortedSet, SortedDict
+
+from adaptive.learner.base_learner import BaseLearner
+
+inf = sys.float_info.max
+
+
+class _IgnoreFirstArgument:
+    """Remove the first argument from the call signature.
+
+    The SequenceLearner's function receives a tuple ``(index, point)``
+    but the original function only takes ``point``.
+
+    This is the same as `lambda x: function(x[1])`, however, that is not
+    pickable.
+    """
+
+    def __init__(self, function):
+        self.function = function
+
+    def __call__(self, index_point, *args, **kwargs):
+        index, point = index_point
+        return self.function(point, *args, **kwargs)
+
+    def __getstate__(self):
+        return self.function
+
+    def __setstate__(self, function):
+        self.__init__(function)
+
+
+class SequenceLearner(BaseLearner):
+    r"""A learner that will learn a sequence.
+
+    This is useful when your problem cannot be formulated in terms of
+    another adaptive learner, but you still want to use Adaptive's
+    routines to run, save, and plot.
```

Comment:
I feel there could be a little more explanation about what the learner does. I'm not sure if "learning a sequence" is very clearn.

Reply:
something that explains that the learner simply gives back the points in the provided sequence when `ask`ed.